spec/features/prison_information_page_spec.rb

```diff
@@ -1,43 +1,74 @@
+        context 'when a prison is not selected' do
+          it 'displays an error message' do
+            click_button 'Continue'
 
-      context "and leaves fields blank" do
-        it "validation messages are present when a prison is not selected" do
-          visit '/prisoner-details'
+            expect(page).to have_css "label[for='prisoner_prison_name'] .validation-message"
+          end
+        end
+
+        context 'when prisoner related form fields are left blank' do
+          let(:prisoner_element_ids) { %w<first_name last_name date_of_birth_3i number> }
 
-          click_button 'Continue'
+          it 'displays error messages for each field' do
+            set_prison_to 'Cardiff'
+            click_button 'Continue'
 
-          expect(page).to have_css("label[for='prisoner_prison_name'] .validation-message")
+            prisoner_element_ids.each do |id|
+              expect(page).to have_css ".validation-error #prisoner_#{id}"
+            end
+          end
         end
 
-        it "validation messages are present" do
-          visit '/prisoner-details'
+        context 'when a prison that is disabled is selected' do
+          let(:a_disabled_prison) { 'Rye Hill' }
 
-          find(:css, ".ui-autocomplete-input").set('Cardiff')
-          click_button 'Continue'
+          it 'displays a message about the prison being disabled' do
+            set_prison_to a_disabled_prison
+            click_button 'Continue'
 
-          expect(page).to have_css(".validation-error #prisoner_first_name")
-          expect(page).to have_css(".validation-error #prisoner_last_name")
-          expect(page).to have_css(".validation-error #prisoner_date_of_birth_3i")
-          expect(page).to have_css(".validation-error #prisoner_number")
+            expect(page).to have_content 'HMP Rye Hill is unable to process online visit requests.'
+          end
         end
-      end
 
-      context "and they fill out all fields" do
-        it "prompts for visitor information" do
-          visit '/prisoner-details'
+        context 'when a prison is coming soon' do
+          let(:a_prison_coming_soon) { 'Hull' }
+
+          it 'displays a message about the prison not being available just yet' do
+            set_prison_to a_prison_coming_soon
+            click_button 'Continue'
+
+            expect(page).to have_content 'HMP Hull isn’t able to process online visit requests yet.'
+          end
+        end
+
+        context 'when a prison has IT issues' do
+          let(:a_prison_with_it_issues) { 'Blantyre House' }
 
-          enter_prisoner_information(flow)
+          it 'displays a message about the prison not being available just yet' do
+            set_prison_to a_prison_with_it_issues
+            click_button 'Continue'
 
-          expect(page).to have_content('Visitor 1')
+            expect(page).
+              to have_content 'HMP Blantyre House is unable to process online visit requests right now.'          end
```

Comment:
The `end` was off screen in my editor :cry: - FP'd it onto the correct line. Well spotted.